### PR TITLE
Rewrote class searcher util

### LINKED ISSUE #6: Fix ClassSearcherUtil
It only works when you run the project from ur ide but not if the code is already compiled

### PR DESCRIPTION
**Pull request by tidyycode**

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -12,16 +12,15 @@ repositories {
...
     implementation project(':api')
-    
-    implementation 'com.google.code.gson:gson:2.11.0'
+    implementation 'org.jetbrains:annotations:24.0.0'
+    implementation 'org.projectlombok:lombok:1.18.30'
+    testImplementation platform('org.junit:junit-bom:5.10.0')
+    testImplementation 'org.junit.jupiter:junit-jupiter'
+    testImplementation 'org.mockito:mockito-core:5.13.0'
+    testImplementation 'org.mockito:mockito-junit-jupiter:5.13.0'
+    implementation 'com.google.code.gson:gson:2.8.8'
```
Comment (Tjorven-Liebe) [inline common/build.gradle]:
```suggestion
    implementation 'org.jetbrains:annotations:24.0.0'
    implementation 'com.google.code.gson:gson:2.8.8'
    implementation 'org.projectlombok:lombok:1.18.30'
    annotationProcessor 'org.projectlombok:lombok:1.18.34'
    
    testImplementation platform('org.junit:junit-bom:5.10.0')
    testImplementation 'org.junit.jupiter:junit-jupiter'
    testImplementation 'org.mockito:mockito-core:5.13.0'
    testImplementation 'org.mockito:mockito-junit-jupiter:5.13.0'
    testAnnotationProcessor 'org.projectlombok:lombok:1.18.34'
```

File: common/src/test/java/de/interguess/javaflow/test/ClassSearchTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+
+    @Test
+    void testException() {
+        // Simulate IOException in ClassPath
+        try (var mockedStatic = mockStatic(ClassPath.class)) {
+            mockedStatic.when(() -> ClassPath.from(mockClassLoader)).thenThrow(new IOException());
+
+            assertThrows(RuntimeException.class, () -> ClassSearcherUtil.listAllClasses(mockClassLoader, "com.example.package"));
```
Comment (Tjorven-Liebe) [inline common/src/test/java/de/interguess/javaflow/test/ClassSearchTest.java:68]: com.example.package?

**Review by Tjorven-Liebe — CHANGES_REQUESTED** [review]

File: common/src/test/java/de/interguess/javaflow/test/ClassSearchTest.java (hunk 2 above)
Comment (tidyycode) [reply common/src/test/java/de/interguess/javaflow/test/ClassSearchTest.java:68]: placeholder; good readability

**Review by JonasDevDE — CHANGES_REQUESTED** [review]
dependencies jetbrains:annotations and projectlombok:lombok doesnt require to be an implementation

**Pull request merged**